src/EPR.Calculator.API/Controllers/CalculatorController.cs

```diff
@@ -172,11 +173,76 @@ public IActionResult GetCalculatorRuns([FromBody] CalculatorRunsParamsDto reques
         }
 
         [HttpGet]
-        [Route("calculatorRuns/{id}")]
-        public IActionResult GetCalculatorRun()
+        [Route("calculatorRuns/{runId}")]
+        public IActionResult GetCalculatorRun(int runId)
         {
-            // TODO: Return the details of a particular run
-            return new OkResult();
+            if (!ModelState.IsValid)
+            {
+                return StatusCode(StatusCodes.Status400BadRequest, ModelState.Values.SelectMany(x => x.Errors));
+            }
+
+            var calculatorRunDetail =
+                (from run in this.context.CalculatorRuns
+                join classification in context.CalculatorRunClassifications
+                    on run.CalculatorRunClassificationId equals classification.Id
+                where run.Id == runId
+                select new
+                {
+                    Run = run,
+                    Classification = classification
+                }).SingleOrDefault();
+
+            
+            if (calculatorRunDetail == null)
+            {
+                return new NotFoundObjectResult($"Unable to find Run Id {runId}");
+            }
+            var calcRun = calculatorRunDetail.Run;
+            var runClassification = calculatorRunDetail.Classification;
+            var runDto = CalcRunMapper.Map(calcRun, runClassification);
+
+            return new ObjectResult(runDto);
+        }
+
+        [HttpPut]
+        [Route("calculatorRuns")]
+        public IActionResult PutCalculatorRunStatus(CalculatorRunStatusUpdateDto runStatusUpdateDto)
```

Comment:
Shall we surround with a try catch and return 500 Internal server error if any other exceptions?

Reply:
Yeah agree, Done the changes now. Thanks